java.util.ServiceConfigurationError: io.smallrye.config.SmallRyeConfigFactory when injecting same configuration parameter in multiple test classes

**Describe the bug**
When injecting a config property annotated with `@Configproperty(name = 'myproperty')` and `@Inject` in two separate Test classes, it will run the test fine on the first class, but on the second class it will throw the following exception:

```
<output of testclass1 truncated>
[INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 9.418 s - in dev.testclass1.ObjectsAzureImplTest
[INFO] Running dev.testclass2.RepositoriesAzureImplTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.001 s <<< FAILURE! - in dev.testclass2.RepositoriesAzureImplTest
[ERROR] dev.testclass2.RepositoriesAzureImplTest  Time elapsed: 0.001 s  <<< ERROR!
org.junit.jupiter.api.extension.TestInstantiationException: Failed to create test instance
Caused by: java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
Caused by: java.lang.reflect.InvocationTargetException
Caused by: java.util.ServiceConfigurationError: io.smallrye.config.SmallRyeConfigFactory: io.quarkus.runtime.configuration.QuarkusConfigFactory not a subtype

2020-03-27 09:07:01,704 INFO  [io.quarkus] (main) Quarkus stopped in 0.033s
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   RepositoriesAzureImplTest » TestInstantiation Failed to create test instance
[INFO] 
[ERROR] Tests run: 11, Failures: 0, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  13.404 s
[INFO] Finished at: 2020-03-27T09:07:02+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test)
```
Disabling one of the two test classes means both classes run fine, so the tests themselves have no error. After removing the `@Inject` and `@Configproperty(name = 'myproperty')`, and hardcoding the value, both tests are running as expected. So the problem is clearly in the injection mechanism.

Note that this behavior was not with Quarkus 1.2 and started after upgrading to Quarkus 1.3

**Expected behavior**
I can inject the `myproperty` set in `applications.properties` across multiple test classes and have my test classes all succeed.

**Actual behavior**
Currently this injection only works for one class, and will fail once the second test class is starting.

**To Reproduce**
Steps to reproduce the behavior:
1. Create two separate test classes and a config parameter in application.properties
2. Inject the parameter in both test classes
3. Watch the first test succeed, and the second one fail. Run them individually and the test cases both succeed.

**Environment (please complete the following information):**
 - Quarkus version or git rev: 1.3.0
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
```
Maven home: /home/jorn/.m2/wrapper/dists/apache-maven-3.6.3-bin/1iopthnavndlasol9gbrbg6bf2/apache-maven-3.6.3
Java version: 11.0.6, vendor: AdoptOpenJDK, runtime: /opt/jdk-11.0.6+10
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "4.19.84-microsoft-standard", arch: "amd64", family: "unix"
```

--- a/integration-tests/main/src/test/java/io/quarkus/it/main/TestInstancePerClass2Test.java
+++ b/integration-tests/main/src/test/java/io/quarkus/it/main/TestInstancePerClass2Test.java
@@ -1,0 +1,21 @@
+package io.quarkus.it.main;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class TestInstancePerClass2Test {
+
+    @ConfigProperty(name = "web-message")
+    String message;
+
+    @Test
+    void testInjectionWorksProperly() {
+        Assertions.assertNotNull(message);
+    }
+}

--- a/integration-tests/main/src/test/java/io/quarkus/it/main/TestInstancePerClassTest.java
+++ b/integration-tests/main/src/test/java/io/quarkus/it/main/TestInstancePerClassTest.java
@@ -1,0 +1,21 @@
+package io.quarkus.it.main;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class TestInstancePerClassTest {
+
+    @ConfigProperty(name = "web-message")
+    String message;
+
+    @Test
+    void testInjectionWorksProperly() {
+        Assertions.assertNotNull(message);
+    }
+}

--- a/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java
+++ b/test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java
@@ -267,6 +267,14 @@ public class QuarkusTestExtension
         if (failedBoot) {
             return result;
         }
+
+        // We do this here as well, because when @TestInstance(Lifecycle.PER_CLASS) is used on a class,
+        // interceptTestClassConstructor is called before beforeAll, meaning that the TCCL will not be set correctly
+        // (for any test other than the first) unless this is done
+        if (runningQuarkusApplication != null) {
+            setCCL(runningQuarkusApplication.getClassLoader());
+        }
+
         initTestState(extensionContext, state);
         return result;
     }

